src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash-es';
+
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+function deleteCategory(allRelationships, category) {
+  delete allRelationships[category];
+  return allRelationships;
+}
+
+function deleteRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+) {
+  const targetRelationshipCategory = get(
+    allRelationships,
+    currentCategory,
+  );
+  const relationshipsWithoutTarget = targetRelationshipCategory.filter(
```

Comment:
so if `get` fails to find anything on line 23, then `null.filter()` will crash the app here. so either (1) you are confident that currentCategory will always exist on the object because we aren't depending on anything external and we can just do 
```
const targetRelationshipCategory = allRelationships[currentCategory];
```
or (2) we can't rely on these things and need to provide a better fallback
```
  const targetRelationshipCategory = get(
    allRelationships,
    currentCategory,
    []
  );
```

Reply:
Good catch. Thanks. Done.